Upgrade to .NET 10.0.

maliming opened the pull request:
Resolve #23383

maliming committed:
Upgrade to .NET 10.0.

maliming committed:
Fix directory reversal and update target frameworks

This may be a SDK problem, I will check it when the RC is released.

maliming committed:
Refactor PackageReference for `coverlet.collector`.

maliming committed:
Update package references to use version 10.0.0-rc.* for consistency across projects

maliming committed:
Remove development mode static assets mapping logic

https://github.com/dotnet/aspnetcore/issues/59673#issuecomment-3259182522

maliming committed:
Update target frameworks to include netstandard2.1 and adjust SDK version in global.json

maliming committed:
Merge branch 'dev' into net10.0

maliming committed:
Merge branch 'dev' into net10.0

maliming committed:
Remove dotnet10 package source from NuGet.Config

maliming committed:
Merge branch 'net10.0' of github.com:abpframework/abp into net10.0

maliming committed:
Refactor exception message construction in test

maliming committed:
Update MongoDB.Driver and improve permission query

maliming committed:
Revert changes to `MongoPermissionGrantRepository`.

maliming committed:
Refactor LINQ queries to use AsEnumerable() for better compatibility with MongoDb Driver.

https://learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/10.0/csharp-overload-resolution#recommended-action

maliming committed:
Update package versions in Directory.Packages.props

Upgraded MongoDB.Driver to 3.5.0, Npgsql.EntityFrameworkCore.PostgreSQL to 10.0.0-rc.1, and MySql.EntityFrameworkCore to 10.0.0-preview to ensure compatibility with latest features and bug fixes.

maliming committed:
Add `BLOB Storing Memory Provider` module.

maliming committed:
Merge branch 'dev' into net10.0

maliming committed:
Remove obsolete features.

https://learn.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/10/razor-runtime-compilation-obsolete
https://learn.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/10/iactioncontextaccessor-obsolete

maliming committed:
Add ABP 10.0 migration guide documentation

hikalkan commented on docs/en/release-info/migration-guides/abp-10-0.md:
```diff
@@ -0,0 +1,67 @@
+    * `Discard`: The event that caused the exception will be discarded and will not be retried.
+* `InboxProcessorRetryBackoffFactor`: The initial retry delay factor (double) used when `InboxProcessorFailurePolicy` is `RetryLater`. The retry delay is calculated as: `delay = InboxProcessorRetryBackoffFactor × 2^retryCount`. Default value is `10`.
+
+See the [Add failure retry policy to InboxProcessor](https://github.com/abpframework/abp/pull/23563) PR for more information.
```
It is not clear why this is in the migration guide. You can write something like "Since this feature brings some new fields in the database tables, you must add a new database migration if you are using Entity Framework Core".

maliming replied:
https://github.com/abpframework/abp/pull/23609/commits/5ff7fca3059454cb8a433d39845775a4a14c3dc3

maliming committed:
Update migration guide for InboxProcessor changes

Clarified the addition of failure retry policy options to AbpEventBusBoxesOptions and noted required database migration due to changes in IncomingEventRecord fields for Entity Framework Core users.

maliming committed:
Update ABP 10.0 migration guide for OpenIddict and Mapperly

maliming committed:
Merge branch 'dev' into net10.0

maliming committed:
Upgrade all NuGet package to latest.

maliming committed:
Fix mongo unit tests.

maliming committed:
Update npm package dependencies and add ncu script

maliming committed:
Refactor image format handling in resizer

maliming committed:
Remove custom WebRootFileProvider initialization

maliming committed:
Enable Pomelo MySQL project and add EF Core Relational

maliming committed:
Workaround for .NET 10 RC1 embedded file provider issue

Adds a temporary fix for a known issue with Microsoft.Extensions.FileProviders.Embedded in .NET 10 RC1, ensuring correct resource name extraction for web content folders. See https://github.com/dotnet/aspnetcore/issues/63719 for details.

maliming committed:
Include wwwroot in embedded file provider folders

Added 'wwwroot' to the list of web content folders in AbpEmbeddedFileProvider to address resource detection issues, following a workaround for a known .NET 10 RC 1 issue.

maliming committed:
Add FodyWeavers config and schema files

maliming committed:
Merge branch 'dev' into net10.0

enisn commented:
There is a compilation error on maui template

```
 error CS5001: Program does not contain a static 'Main' method suitable for an entry point
    C:\P\abp\templates\maui\src\MyCompanyName.MyProjectName\App.xaml.cs(9,3): warning CS0618: 'Application.MainPage.set' is obsolete: 'This property is deprecated. Initialize your application by overriding Application.CreateWindow rather than setting MainPage. To modify the root page in an active application, use Windows[0].Page for applications with a single window. For applications with multiple windows, use Application.Windows to identify and update the root page on the correct window.  Additionally, each element features a Window property, accessible when it's part of the current window.'
```

maliming committed:
Update target frameworks in MAUI project file

maliming commented:
> There is a compilation error on maui template

Fixed by https://github.com/abpframework/abp/pull/23609/commits/67c708f4361cb82f5112fbe725f18b96ed6017c0

maliming committed:
Document cache error hiding behavior in development

maliming committed:
Merge branch 'dev' into net10.0
